pkg/volume/local/local.go

```diff
@@ -208,6 +207,26 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return nil
 	}
 
+	oldMount, err := mount.IsAlreadyMounted(dir)
+	if err != nil {
+		glog.Errorf("failed to check if disk %s has already been mounted(%v)", dir, err)
+		return err
+	}
+
+	if len(oldMount) > 0 {
```

Comment:
I think it also should protect from that case as well. Here is the logic I am using.
1. Check if mount table has already a reference to volume name. if not, business as usual
2. If yes. Check Gid for found old mounted volume and compare with the one specified in pod spec. If the same, then proceed  with mouting, if not fail mount.
It seems this logic covers all cases. Please clarify if you think it does not.